### PR TITLE
Now the average monthly spendings for this year only considers expenses for current year.

### PR DESCRIPTION
**Pull request by HKanwal**
## Description
Now the average monthly spendings for this year only considers expenses for current year.

## Related Issue
- [ ] This pull request relates to at least one particular issue

  
  - [issue name](https://github.com/ps-toronto-team-4/.github/issues/ISSUENUMBER)

## Checks
- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information

None

### CODE REVIEW

File: screens/HomeScreen.tsx [hunk 1]
```diff
@@ -97,7 +97,9 @@ export default function HomeScreen({ navigation }: RootTabScreenProps<'Home'>) {
     const averageData = useMemo(() => {
         let yearAvg = 0;
         if (avgData?.monthsTotals.__typename === 'MonthsTotals' && avgData.monthsTotals.averageSpent) {
-            yearAvg = avgData.monthsTotals.averageSpent;
+            const applicableMonthData = avgData.monthsTotals.byMonth.filter(x => x.year === year).map(x => x.amountSpent);
+            console.log(applicableMonthData.length);
```
Comment (1mozolacal) [inline screens/HomeScreen.tsx:101]: Remove this

Comment (1mozolacal) [reply screens/HomeScreen.tsx:101]: The log I am talking about just to clarify

File: screens/HomeScreen.tsx [hunk 2]
```diff
@@ -97,7 +97,9 @@ export default function HomeScreen({ navigation }: RootTabScreenProps<'Home'>) {
...
+            yearAvg = applicableMonthData.reduce((a, b) => a + b, 0) / applicableMonthData.length;
```
Comment (1mozolacal) [inline screens/HomeScreen.tsx:102]: Add a check for zero length.

**Review by 1mozolacal — CHANGES_REQUESTED** [review]

File: screens/HomeScreen.tsx (hunk 2 above)
Comment (HKanwal) [reply screens/HomeScreen.tsx]: Nice catch

**Pull request merged**